hooks/persistence-defectdojo/src/main/java/io/securecodebox/persistence/DefectDojoPersistenceProvider.java

```diff
@@ -46,17 +49,27 @@ public static void main(String[] args) throws Exception {
     scan.validate();
 
     var config = DefectDojoConfig.fromEnv();
-
-    LOG.info("Downloading Scan Report (RawResults)");
-    var rawResults = s3Service.downloadRawResults(persistenceProviderConfig.getRawResultDownloadUrl());
-    LOG.info("Finished Downloading Scan Report (RawResults)");
-    LOG.debug("RawResults: {}", rawResults);
+    LOG.info("Downloading Scan Result");
+    String downloadUrl;
+    String scanType = scan.getSpec().getScanType();
+    ScanNameMapping scanNameMapping = ScanNameMapping.bySecureCodeBoxScanType(scanType);
+    if (scanNameMapping == ScanNameMapping.GENERIC) {
+      LOG.debug("No explicit Parser specified for ScanType {}, using Findings JSON Scan Result", scanType);
+      downloadUrl = persistenceProviderConfig.getFindingDownloadUrl();
+    } else {
+      LOG.debug("Explicit Parser is specified for ScanType {}, using Raw Scan Result", scanNameMapping.scanType);
+      downloadUrl = persistenceProviderConfig.getRawResultDownloadUrl();
+    }
+    var scanResults = s3Service.downloadFile(downloadUrl);
+    LOG.info("Finished Downloading Scan Result");
+    LOG.debug("Scan Result: {}", scanResults);
 
     LOG.info("Uploading Findings to DefectDojo at: {}", config.getUrl());
 
     var defectdojoImportStrategy = new VersionedEngagementsStrategy();
     defectdojoImportStrategy.init(config);
-    var defectDojoFindings = defectdojoImportStrategy.run(scan, rawResults);
+    var scanResultFile = new ScanFile(scanResults, FilenameUtils.getName(new URL(downloadUrl).getPath()));
```

Comment:
You should be able to get the filename of the raw results file directly from the Scan status, instead of parsing it from the downloadUrl:

```suggestion
    var scanResultFile = new ScanFile(scanResults, scan.getStatus().getRawResultFile());
```